### PR TITLE
general refactor & deprecating CastTable

### PR DESCRIPTION
**Pull request by hooahest**
This is a general refactor of some code I came across. Some variable renaming, a 'for' loop that's replaced with ES6, and a component that doesn't seem to be used at anyplace in the code so I wrote it as deprecated (better to delete?)

### CODE REVIEW

File: src/components/Match/CastTable.jsx [hunk 1]
```diff
@@ -34,19 +34,19 @@ const castsColumns = strings => [{
   displayFn: (row, col, field) => field || '-',
 }];
 
-const getCastArray = (pm) => {
+const getCastArray = (player) => {
   // Get from ability_uses, item_uses
   const resultArray = [];
   const targets = ['ability_uses', 'item_uses'];
```
Comment (bdbch) [inline src/components/Match/CastTable.jsx]: Should we also rename this? targets isn't really describing what this is used for. Should be called `castKeys` or something I guess?

Comment (hooahest) [reply src/components/Match/CastTable.jsx]: targetedCastTypes? kinda long but describes it well

Comment (hooahest) [reply src/components/Match/CastTable.jsx]: though I'm also okay with castKeys

**Comment by hooahest** [comment]
retest this please

File: src/components/Match/CastTable.jsx (hunk 1 above)
Comment (bdbch) [reply src/components/Match/CastTable.jsx]: I'm fine with longer names as long as it makes sense and is easy to understand :)

**Comment by bdbch** [comment]
> hm, I'm not sure why travis failed
> 
> image
> 
> should I just rerun it? how do I do that?

I think it's an issue with the CI, not your code.

File: src/components/Match/CastTable.jsx [hunk 2]
```diff
@@ -55,16 +55,17 @@ const getCastArray = (pm) => {
   return resultArray;
 };
 
+// Deprecated & unused
```
Comment (howardchung) [inline src/components/Match/CastTable.jsx]: If it's not used, I'd just get rid of it. We can always pull it back out of the history.

File: src/components/Match/matchColumns.jsx [hunk 3]
```diff
@@ -1236,10 +1236,8 @@ export default (strings) => {
         totalDuration.push(duration);
       }
     });
-    let total = 0;
-    for (let i = 0; i < totalDuration.length; i += 1) {
-      total += totalDuration[i];
-    }
+
+    const total = totalDuration.reduce((a, b) => a + b, 0);
```
Comment (howardchung) [inline src/components/Match/matchColumns.jsx]: I'm fine with this, but the original might be a little easier to read. Not everyone is familiar with the .reduce function.

Comment (hooahest) [reply src/components/Match/matchColumns.jsx]: True. This is as basic as a reduce function gets though, so even if someone is not familiar with it, he can probably still understand it (also probably a good way to learn it?)

File: src/components/Match/CastTable.jsx (hunk 2 above)
Comment (hooahest) [reply src/components/Match/CastTable.jsx]: deleted it

**Pull request merged**